pkg/buildpod/generator.go

```diff
@@ -167,3 +176,49 @@ func parseCNBID(image ggcrv1.Image, env string) (int64, error) {
 	}
 	return strconv.ParseInt(v, 10, 64)
 }
+
+func (g *Generator) calculateHomogenousWindowsNodeTaints() ([]v1.Taint, error) {
+	windowsNodes, err := g.K8sClient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: "kubernetes.io/os=windows"})
+	if err != nil {
+		return nil, err
+	}
+
+	nodeList := windowsNodes.Items
+	if len(nodeList) == 0 {
+		return []v1.Taint{}, nil
+	}
+
+	taints := nodeList[0].Spec.Taints
+	sort.Slice(taints, func(i, j int) bool {
+		return taints[i].Key < taints[j].Key
+	})
+
+	for _, node := range nodeList[1:] {
+		taintsToCompare := node.Spec.Taints
+		sort.Slice(taintsToCompare, func(i, j int) bool {
+			return taintsToCompare[i].Key < taintsToCompare[j].Key
+		})
```

Comment:
As an optimization, maybe we can move this second sort to be inside the `func taintsEqua(`, after the size check.

Reply:
i had that originally, but i changed it because even though its slightly less performant, i felt like it was a little weird to be sorting in the equals function, especially since i was only sorting one slice